hashintel/hash
# H-1264: Check / fix existing block functionality

## 🌟 What is the purpose of this PR?




This PR fixes several blocks by upgrading to the latest canary release of the `@blockprotocol/graph` package, which ensures that blocks making use of the non-temporal BP methods from that package receive non-temporal returns from those methods, **even when the block as been a temporal subgraph**.

It also fixes other bugs that were spotted in the `Address` block:
- updates the `graphService` callback functions that modify data to all trigger a refetch of the subgraph that is passed to the block, so that when the `Address` block uploads an image it can obtain this from the subgraph provided to it
- updates logic which determines how the `Address` block fetches the image of an address via the `serviceModule`, so that it doesn't attempt to fetch this multiple times for the same address

## 🔗 Related links





- H-1264

## 🔍 What does this change?




See description.

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:

- [x] modifies multiple **blocks** that will need publishing via GitHub action once merged

### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



None

## ❓ How to test this?



1. run the `Address` block locally, and load it into the HASH App and try to use it. Check that it is only performing the expected number of requests to the `/external-service-method` endpoint via the network tab.
2. Run the `AI Chat` block locally, and load it into the HASH App and start using it. Refresh the page to ensure no runtime error is thrown by the `@blockprotocol/graph` methods it uses.

## 📹 Demo



<img width="1089" alt="image" src="https://github.com/hashintel/hash/assets/42802102/fe6e13b8-47e5-4ad7-b5aa-a610daccd190">

<img width="1073" alt="image" src="https://github.com/hashintel/hash/assets/42802102/25513ded-bef7-4142-90bc-dd3bf306fdc7">
